src/components/MAPI/organizations/__tests__/OrganizationDetail.tsx

```diff
@@ -0,0 +1,171 @@
+  });
+
+  afterEach(() => {
+    cache.clear();
+  });
+
+  it('renders without crashing', async () => {
+    render(getComponent({}));
+
+    await waitForElementToBeRemoved(screen.getByTitle('Loading...'));
+  });
+
+  it('provides the ability to delete an organization', async () => {
+    nock(window.config.mapiEndpoint)
+      .get('/apis/security.giantswarm.io/v1alpha1/organizations/org1/')
+      .reply(StatusCodes.Ok, securityv1alpha1Mocks.getOrganizationByName);
+    nock(window.config.mapiEndpoint)
+      .delete('/apis/security.giantswarm.io/v1alpha1/organizations/org1/')
+      .reply(StatusCodes.Ok, securityv1alpha1Mocks.getOrganizationByName);
+
+    nock(window.config.mapiEndpoint)
+      .post('/apis/authorization.k8s.io/v1/selfsubjectaccessreviews/')
+      .reply(
+        StatusCodes.Ok,
+        authorizationv1Mocks.selfSubjectAccessReviewCanListOrgs
+      );
+    nock(window.config.mapiEndpoint)
+      .get('/apis/security.giantswarm.io/v1alpha1/organizations/')
+      .reply(StatusCodes.Ok, securityv1alpha1Mocks.getOrganizationListResponse);
+
+    render(getComponent({}));
+
+    await waitForElementToBeRemoved(screen.getByTitle('Loading...'));
+
+    const deleteButton = screen.getByText('Delete Organization');
+    expect(deleteButton).toBeInTheDocument();
+
+    fireEvent.click(deleteButton);
+    fireEvent.click(screen.getByText('Yes, delete it'));
+
+    expect(screen.getByRole('progressbar')).toBeInTheDocument();
+    expect(deleteButton).toBeDisabled();
+
+    expect(
+      await screen.findByText('Organization org1 deleted successfully.')
+    ).toBeInTheDocument();
+  });
+
+  it('displays an error if deleting an organization fails', async () => {
+    nock(window.config.mapiEndpoint)
+      .get('/apis/security.giantswarm.io/v1alpha1/organizations/org1/')
+      .reply(StatusCodes.Ok, securityv1alpha1Mocks.getOrganizationByName);
+    nock(window.config.mapiEndpoint)
+      .delete('/apis/security.giantswarm.io/v1alpha1/organizations/org1/')
+      .reply(StatusCodes.InternalServerError, {
+        apiVersion: 'v1',
+        kind: 'Status',
+        message: 'There was a huge problem.',
+        status: metav1.K8sStatuses.Failure,
+        reason: metav1.K8sStatusErrorReasons.InternalError,
+        code: StatusCodes.InternalServerError,
+      });
+
+    render(getComponent({}));
+
+    await waitForElementToBeRemoved(screen.getByTitle('Loading...'));
+
+    const deleteButton = screen.getByText('Delete Organization');
+    expect(deleteButton).toBeInTheDocument();
+
+    fireEvent.click(deleteButton);
+    fireEvent.click(screen.getByText('Yes, delete it'));
+
+    expect(screen.getByRole('progressbar')).toBeInTheDocument();
+    expect(deleteButton).toBeDisabled();
+
+    expect(
+      await screen.findByText('Could not delete organization org1:')
+    ).toBeInTheDocument();
+    expect(screen.getByText('There was a huge problem.')).toBeInTheDocument();
+  });
+
+  it('can cancel deletion', async () => {
```

Comment:
I think this description is a bit wrong? 

Do we really support cancelling a inflight request? 

I think this is more like just hiding the confirmation right? And if so this test implies it's possible to click cancel after the delete request is in progress. (maybe not possible for a human since I see it is being hidden when you confirm the deletion)

but yeah.. maybe just line 166 needs to be removed

Reply:
This test is just for the ability to press `Cancel` and have the dialog go away (essentially cancelling deletion).

So the user presses the `Delete Organization` button, a confirm dialog pops up, and if they press `Cancel`, we expect the dialog to go away. Without line `166`, the dialog wouldn't show up.

Reply:
oh yeahhh I totally misunderstood the test. my bad :)